### PR TITLE
feat: add new design and mnemonic screen

### LINKED ISSUE #26: Edit permissions
Currently it is not possible to edit and update the permissions after a connection is created. 

It should be possible to update the budget.

I am not sure about changing the features because that might break the app?

### LINKED ISSUE #58: Move mnemonic backup out of create wallet onboarding
image

### PR DESCRIPTION
**Pull request by im-adithya**
- Implements the new designs from Figma
- Adds Mnemonic Screen
- Adds feature to edit Permissions

Fixes #26
Fixes #58

⚠️ Adds `nextBackupReminder` to `user_configs` which stores the date on which the mnemonic is last backed up
⚠️ `InfoResponse` now carries `showBackupReminder` which is a `boolean`

### CODE REVIEW

File: frontend/src/screens/Mnemonic.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      handleRequestError("Failed to store back up info", error);
+    }
+  }
+
+  return (
+    <>
+      {!decryptedMnemonic ? (
+        <Container>
+          <p className="font-light text-center text-md leading-relaxed dark:text-neutral-400 mb-14">
+            Enter your password to decrypt mnemonic
```
Comment (rolznz) [inline frontend/src/screens/Mnemonic.tsx:76]:
```suggestion
            Enter your unlock password to continue
```

File: frontend/src/components/SuggestedApps.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
+import { Link } from "react-router-dom";
+
+import alby from "src/assets/suggested/alby.png";
```
Comment (rolznz) [inline frontend/src/components/SuggestedApps.tsx:3]: could you rename the `suggested` folder to `suggested-apps` so it's more clear? should it be `recommended-apps`?

File: api.go [hunk 3]
```diff
@@ -310,14 +313,47 @@ func (api *API) GetMempoolLightningNode(pubkey string) (interface{}, error) {
...
 	unlockPasswordCheck, _ := api.svc.cfg.Get("UnlockPasswordCheck", "")
 	info.SetupCompleted = unlockPasswordCheck != ""
 	info.Running = api.svc.lnClient != nil
 	info.BackendType = backendType
-	return &info
+
+	if info.BackendType == LNDBackendType {
+		info.IsMnemonicBackupDone = true // because otherwise it shows the popup
+	} else {
+		isMnemonicBackupDone, _ := api.svc.cfg.Get("IsMnemonicBackupDone", "")
```
Comment (rolznz) [inline api.go:327]: `IsMnemonicBackupDone` is a time field, right, so that's a bad name?

@bumi suggested a nice idea, what about `NextBackupReminder` which stores the next time the reminder will be shown

File: api.go [hunk 4]
```diff
@@ -310,14 +313,47 @@ func (api *API) GetMempoolLightningNode(pubkey string) (interface{}, error) {
...
+		}
+		currentTime := time.Now()
+		sixMonthsAgo := currentTime.AddDate(0, -6, 0)
+		info.IsMnemonicBackupDone = !(parsedTime.IsZero() || parsedTime.Before(sixMonthsAgo))
+	}
+
+	return &info, nil
+}
+
+func (api *API) GetMnemonic() *models.MnemonicResponse {
```
Comment (rolznz) [inline api.go:345]:
```suggestion
func (api *API) GetEncryptedMnemonic() *models.EncryptedMnemonicResponse {
```

File: api.go [hunk 5]
```diff
@@ -310,14 +313,47 @@ func (api *API) GetMempoolLightningNode(pubkey string) (interface{}, error) {
...
+	resp := models.MnemonicResponse{}
+	mnemonic, _ := api.svc.cfg.Get("Mnemonic", "")
+	resp.Mnemonic = mnemonic
+	return &resp
+}
+
+func (api *API) BackupMnemonic() error {
```
Comment (rolznz) [inline api.go:352]:
```suggestion
func (api *API) SetNextBackupReminder(nextBackupReminder Date) error {
```

File: api.go [hunk 6]
```diff
@@ -327,6 +363,8 @@ func (api *API) Start(startRequest *models.StartRequest) error {
 func (api *API) Setup(setupRequest *models.SetupRequest) error {
 	api.svc.cfg.SavePasswordCheck(setupRequest.UnlockPassword)
 
+	// update mnemonic check
+	api.svc.cfg.SetUpdate("IsMnemonicBackupDone", setupRequest.IsMnemonicBackupDone, "")
```
Comment (rolznz) [inline api.go:367]: why is this needed?

Comment (im-adithya) [reply api.go:367]: In the case of imports, the user already has it backed up right, so we don't show it till 6 months

File: api.go (hunk 3 above)
Comment (im-adithya) [reply api.go:327]: Yup that sounds great :raised_hands:

File: frontend/src/components/ConnectionItem.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+import alby from "src/assets/suggested/alby.png";
+
+type Props = {
+  app: App;
+  csrf?: string;
+  onDelete: (nostrPubkey: string) => void;
+};
+
+export default function ConnectionItem({ app, csrf, onDelete }: Props) {
```
Comment (rolznz) [inline frontend/src/components/ConnectionItem.tsx:23]: We currently use the terminology "App", so what about `AppCard`?

File: frontend/src/components/ConnectionItem.tsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+      onDelete(app.nostrPubkey);
+      toast.success("App disconnected");
+    } catch (error) {
+      await handleRequestError("Failed to delete app", error);
+    } finally {
+      setShowPopup(false);
+    }
+  };
+
+  const Popup = () => (
```
Comment (rolznz) [inline frontend/src/components/ConnectionItem.tsx:48]: We could use a common component here. With shadcn-ui we can use: https://ui.shadcn.com/docs/components/alert-dialog

Comment (rolznz) [reply frontend/src/components/ConnectionItem.tsx:48]: But I think it's ok for now. I guess this should be done in a follow-up PR

File: frontend/src/components/SuggestedApps.tsx [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+  return (
+    <>
+      <h2 className="text-center font-medium px-8 sm:px-0">
+        Use NWC to connect your wallet to any of apps below
+      </h2>
+
+      <div className="my-8 grid sm:grid-cols-2 md:grid-cols-3 gap-4">
+        {suggestedApps.map((app, index) => (
+          <SuggestedApp
+            key={index}
```
Comment (rolznz) [inline frontend/src/components/SuggestedApps.tsx:166]: you could use the app title as a key, it should be unique. The index option is kind of a last resort/hack

File: frontend/src/hooks/useMnemonic.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import useSWR from "swr";
+
+import { swrFetcher } from "src/utils/swr";
+import { MnemonicResponse } from "src/types";
+
+export function useMnemonic() {
```
Comment (rolznz) [inline frontend/src/hooks/useMnemonic.ts:6]:
```suggestion
export function useEncryptedMnemonic() {
```

File: api.go [hunk 11]
```diff
@@ -310,14 +313,47 @@ func (api *API) GetMempoolLightningNode(pubkey string) (interface{}, error) {
...
+		info.IsMnemonicBackupDone = !(parsedTime.IsZero() || parsedTime.Before(sixMonthsAgo))
+	}
+
+	return &info, nil
+}
+
+func (api *API) GetMnemonic() *models.MnemonicResponse {
+	resp := models.MnemonicResponse{}
+	mnemonic, _ := api.svc.cfg.Get("Mnemonic", "")
+	resp.Mnemonic = mnemonic
```
Comment (rolznz) [inline api.go:348]: I don't think this is enough. We should encrypt the payload using public key encryption (both when setting up the app/importing and when returning it here)

File: http_service.go [hunk 12]
```diff
@@ -54,6 +54,8 @@ func (httpSvc *HttpService) RegisterSharedRoutes(e *echo.Echo) {
 
 	e.GET("/api/csrf", httpSvc.csrfHandler)
 	e.GET("/api/info", httpSvc.infoHandler)
+	e.GET("/api/mnemonic", httpSvc.mnemonicHandler)
+	e.PATCH("/api/mnemonic", httpSvc.mnemonicBackupHandler)
```
Comment (rolznz) [inline http_service.go:58]: this route name seems wrong, it doesn't backup anything

File: http_service.go [hunk 13]
```diff
@@ -54,6 +54,8 @@ func (httpSvc *HttpService) RegisterSharedRoutes(e *echo.Echo) {
 
 	e.GET("/api/csrf", httpSvc.csrfHandler)
 	e.GET("/api/info", httpSvc.infoHandler)
+	e.GET("/api/mnemonic", httpSvc.mnemonicHandler)
```
Comment (rolznz) [inline http_service.go:57]: These endpoints need to be protected by the `authMiddleware` :exclamation:

File: frontend/src/screens/setup/SetupMnemonic.tsx [hunk 14]
```diff
@@ -15,16 +11,9 @@ import toast from "src/components/Toast";
...
   const setupStore = useSetupStore();
-  const params = new URLSearchParams(search);
-  const isNew = params.get("wallet") === "new";
 
-  const [mnemonic, setMnemonic] = useState<string>(
-    setupStore.nodeInfo.mnemonic ||
-      (isNew ? bip39.generateMnemonic(wordlist, 128) : "")
-  );
-  const [backedUp, isBackedUp] = useState<boolean>(false);
+  const [mnemonic, setMnemonic] = useState<string>("");
```
Comment (rolznz) [inline frontend/src/screens/setup/SetupMnemonic.tsx:16]: Can you rename the React component and filename to be `ImportMnemonic`?

File: frontend/src/screens/Mnemonic.tsx [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+    try {
+      await request("/api/mnemonic", {
+        method: "PATCH",
+        headers: {
+          "X-CSRF-Token": csrf,
+          "Content-Type": "application/json",
+        },
+      });
+      navigate("/");
+      toast.success("Mnemonic Backed Up!");
```
Comment (rolznz) [inline frontend/src/screens/Mnemonic.tsx:65]: Mnemonic is the technical terminology. Can you use the user-facing terminology here?

File: frontend/src/screens/Mnemonic.tsx [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+
+    try {
+      await request("/api/mnemonic", {
+        method: "PATCH",
+        headers: {
+          "X-CSRF-Token": csrf,
+          "Content-Type": "application/json",
+        },
+      });
+      navigate("/");
```
Comment (rolznz) [inline frontend/src/screens/Mnemonic.tsx:64]: Before navigating we should mutate the info endpoint so it doesn't show the reminder again

Comment (rolznz) [reply frontend/src/screens/Mnemonic.tsx:64]: See `await refetchInfo();`

File: frontend/src/screens/apps/AppsList.tsx [hunk 17]
```diff
@@ -1,81 +1,103 @@
...
-                className="bg-white dark:bg-surface-02dp cursor-pointer hover:bg-purple-50 dark:hover:bg-surface-16dp"
+      <div className="grid sm:grid-cols-2 md:grid-cols-3 gap-4">
+        {!apps.length && (
+          <div className="rounded-2xl bg-indigo-50 border border-indigo-200 flex flex-col justify-between">
+            <div className="p-4 h-full border-b border-indigo-200">
+              <h2 className="font-medium text-indigo-600 mb-2">
+                Create a New Connection
+              </h2>
+              <p className="text-sm text-indigo-600">
+                Click "Add a connection" to create one and use it in your
```
Comment (rolznz) [inline frontend/src/screens/apps/AppsList.tsx:46]: It's weird we tell them to `Click "Add a connection"`

File: frontend/src/screens/apps/AppsList.tsx [hunk 18]
```diff
@@ -1,81 +1,103 @@
...
+                Back up your recovery phrase!
+              </h2>
+              <p className="text-sm text-orange-700">
+                Not backing up your key might result in loosing access to your
+                funds.
+              </p>
+            </div>
+            <div className="py-3 px-4 flex items-center gap-4">
+              <div
+                onClick={() => setShowBackupPrompt(false)}
```
Comment (rolznz) [inline frontend/src/screens/apps/AppsList.tsx:73]: Should we remind in X weeks instead? (possible by passing the next backup reminder as a parameter to the API endpoint)

File: frontend/src/screens/apps/AppsList.tsx [hunk 19]
```diff
@@ -1,81 +1,103 @@
...
-            )}
-
-            {apps.map((app, index) => (
-              <tr
-                key={index}
-                onClick={() => navigate(`/apps/${app.nostrPubkey}`)}
-                className="bg-white dark:bg-surface-02dp cursor-pointer hover:bg-purple-50 dark:hover:bg-surface-16dp"
+      <div className="grid sm:grid-cols-2 md:grid-cols-3 gap-4">
+        {!apps.length && (
+          <div className="rounded-2xl bg-indigo-50 border border-indigo-200 flex flex-col justify-between">
```
Comment (rolznz) [inline frontend/src/screens/apps/AppsList.tsx:40]: I think all these should use a `Card` component. Again this should be possible with shadcn-ui

File: frontend/src/screens/setup/SetupMnemonic.tsx [hunk 20]
```diff
@@ -36,7 +25,21 @@ export function SetupMnemonic() {
...
+      const hours = String(date.getHours()).padStart(2, "0");
+      const minutes = String(date.getMinutes()).padStart(2, "0");
+      const seconds = String(date.getSeconds()).padStart(2, "0");
+
+      return `${year}-${month}-${day} ${hours}:${minutes}:${seconds}`;
+    }
+
+    setupStore.updateNodeInfo({
+      mnemonic,
+      isMnemonicBackupDone: dateToString(new Date()),
```
Comment (rolznz) [inline frontend/src/screens/setup/SetupMnemonic.tsx:41]: Is it possible to use `new Date().toISOString()` and remove the custom method? we could also use date-fns to easily add time to a date

File: frontend/src/utils/aesgcm.ts [hunk 21]
```diff
@@ -0,0 +1,99 @@
+import { hash } from "argon2-wasm-esm";
+
+async function deriveKey(
```
Comment (rolznz) [inline frontend/src/utils/aesgcm.ts:3]: where did you get this code from? can you add a link?

Comment (im-adithya) [reply frontend/src/utils/aesgcm.ts:3]: It's https://github.com/imyelo/argon2-wasm-esm docs, should I add that too?

File: frontend/src/screens/apps/AppsList.tsx (hunk 19 above)
Comment (im-adithya) [reply frontend/src/screens/apps/AppsList.tsx:41]: Leaving as follow-up

File: models/api/api.go [hunk 22]
```diff
@@ -76,10 +84,15 @@ type User struct {
...
 type InfoResponse struct {
-	BackendType    string `json:"backendType"`
-	SetupCompleted bool   `json:"setupCompleted"`
-	Running        bool   `json:"running"`
-	Unlocked       bool   `json:"unlocked"`
+	BackendType        string `json:"backendType"`
+	SetupCompleted     bool   `json:"setupCompleted"`
+	Running            bool   `json:"running"`
+	Unlocked           bool   `json:"unlocked"`
+	NextBackupReminder bool   `json:"NextBackupReminder"`
```
Comment (rolznz) [inline models/api/api.go:91]: Could you rename this field? should it be something like `showBackupReminder`?

Also the json should be lower case

File: api.go [hunk 23]
```diff
@@ -529,14 +618,45 @@ func (api *API) NewWrappedInvoice(request *models.NewWrappedInvoiceRequest) (*mo
...
 	info := models.InfoResponse{}
 	backendType, _ := api.svc.cfg.Get("LNBackendType", "")
 	unlockPasswordCheck, _ := api.svc.cfg.Get("UnlockPasswordCheck", "")
 	info.SetupCompleted = unlockPasswordCheck != ""
 	info.Running = api.svc.lnClient != nil
 	info.BackendType = backendType
-	return &info
+
+	if info.BackendType == config.LNDBackendType {
+		info.NextBackupReminder = true // because otherwise it shows the popup
```
Comment (rolznz) [inline api.go:630]: This field name is confusing. What about `info.ShowBackupReminder = false;`

File: api.go [hunk 24]
```diff
@@ -529,14 +618,45 @@ func (api *API) NewWrappedInvoice(request *models.NewWrappedInvoiceRequest) (*mo
...
+	} else {
+		NextBackupReminder, _ := api.svc.cfg.Get("NextBackupReminder", "")
```
Comment (rolznz) [inline api.go:632]: this is a field, can you change it to camelcase?

Comment (rolznz) [reply api.go:632]: only the field, not the config key

File: api.go [hunk 25]
```diff
@@ -529,14 +618,45 @@ func (api *API) NewWrappedInvoice(request *models.NewWrappedInvoiceRequest) (*mo
...
+		if NextBackupReminder != "" {
+			parsedTime, err = time.Parse(time.RFC3339, NextBackupReminder)
+			if err != nil {
+				api.svc.Logger.Errorf("Error parsing time: %v", err)
+				return nil, err
+			}
+		}
+		currentTime := time.Now()
+		sixMonthsAgo := currentTime.AddDate(0, -6, 0)
+		info.NextBackupReminder = !(parsedTime.IsZero() || parsedTime.Before(sixMonthsAgo))
```
Comment (rolznz) [inline api.go:644]: If changed to `info.ShowBackupReminder` you should be able to remove the `!`

File: api.go (hunk 11 above)
Comment (rolznz) [reply api.go:653]: This can be moved to a new issue, can you make one after merging?

File: api.go [hunk 26]
```diff
@@ -546,6 +666,8 @@ func (api *API) Start(startRequest *models.StartRequest) error {
 func (api *API) Setup(setupRequest *models.SetupRequest) error {
 	api.svc.cfg.SavePasswordCheck(setupRequest.UnlockPassword)
 
+	// update mnemonic check
+	api.svc.cfg.SetUpdate("NextBackupReminder", setupRequest.NextBackupReminder, "")
```
Comment (rolznz) [inline api.go:670]: Why do we set an empty value here? it should not be needed. You can only go through the setup once

File: frontend/src/screens/setup/SetupNode.tsx [hunk 27]
```diff
@@ -21,13 +23,16 @@ export function SetupNode() {
   async function handleSubmit(data: object) {
     setupStore.updateNodeInfo({
       backendType,
+      NextBackupReminder: "0001-01-01T00:00:00Z",
```
Comment (rolznz) [inline frontend/src/screens/setup/SetupNode.tsx:26]: please make it camel case

Comment (rolznz) [reply frontend/src/screens/setup/SetupNode.tsx:26]: does this even need to be passed to the setup though? if there is no value saved, then it will default to showing a reminder

File: models/api/api.go [hunk 28]
```diff
@@ -50,6 +57,7 @@ type SetupRequest struct {
 	// Breez / Greenlight
 	Mnemonic             string `json:"mnemonic"`
 	GreenlightInviteCode string `json:"greenlightInviteCode"`
+	NextBackupReminder   string `json:"NextBackupReminder"`
```
Comment (rolznz) [inline models/api/api.go:60]: I don't  think this is used, can we remove it?

File: frontend/src/components/AppCard.tsx [hunk 29]
```diff
@@ -0,0 +1,136 @@
...
+  app: App;
+  csrf?: string;
+  onDelete: (nostrPubkey: string) => void;
+};
+
+export default function AppCard({ app, csrf, onDelete }: Props) {
+  const navigate = useNavigate();
+  const [showPopup, setShowPopup] = React.useState(false); // State to control popup visibility
+
+  const handleDelete = async () => {
```
Comment (rolznz) [inline frontend/src/components/AppCard.tsx:25]: we can delete in two places now, can this function be moved to a hook e.g. `useDeleteApp`?

File: frontend/src/components/SuggestedApps.tsx [hunk 30]
```diff
@@ -0,0 +1,176 @@
...
+import zapplanner from "src/assets/suggested-apps/zapplanner.png";
+import zapplepay from "src/assets/suggested-apps/zapple-pay.png";
+import lume from "src/assets/suggested-apps/lume.png";
+import bc from "src/assets/suggested-apps/bitcoin-connect.png";
+import kiwi from "src/assets/suggested-apps/kiwi.png";
+import zappybird from "src/assets/suggested-apps/zappy-bird.png";
+import nostur from "src/assets/suggested-apps/nostur.png";
+import wherostr from "src/assets/suggested-apps/wherostr.png";
+
+const suggestedApps = [
```
Comment (rolznz) [inline frontend/src/components/SuggestedApps.tsx:22]: can you make a type
const `suggestedApps: SuggestedApp[] = ...`

Comment (rolznz) [reply frontend/src/components/SuggestedApps.tsx:22]: I see the type exists as `Props`

File: frontend/src/screens/BackupMnemonic.tsx [hunk 31]
```diff
@@ -0,0 +1,170 @@
...
+  };
+
+  async function onSubmit(e: React.FormEvent) {
+    e.preventDefault();
+    if (!csrf) {
+      throw new Error("No CSRF token");
+    }
+
+    try {
+      await request("/api/mnemonic", {
```
Comment (rolznz) [inline frontend/src/screens/BackupMnemonic.tsx:59]: can you please change the route here. We are not "patching the mnemonic". I think I already added a comment last time

File: http_service.go [hunk 32]
```diff
@@ -54,8 +55,11 @@ func (httpSvc *HttpService) RegisterSharedRoutes(e *echo.Echo) {
 	authMiddleware := httpSvc.validateUserMiddleware
 	e.GET("/api/apps", httpSvc.appsListHandler, authMiddleware)
 	e.GET("/api/apps/:pubkey", httpSvc.appsShowHandler, authMiddleware)
+	e.PATCH("/api/apps/:pubkey", httpSvc.appsUpdateHandler, authMiddleware)
 	e.DELETE("/api/apps/:pubkey", httpSvc.appsDeleteHandler, authMiddleware)
 	e.POST("/api/apps", httpSvc.appsCreateHandler, authMiddleware)
+	e.GET("/api/mnemonic", httpSvc.mnemonicHandler, authMiddleware)
```
Comment (rolznz) [inline http_service.go:61]: what about `/api/encrypted-mnemonic`?

File: http_service.go [hunk 33]
```diff
@@ -54,8 +55,11 @@ func (httpSvc *HttpService) RegisterSharedRoutes(e *echo.Echo) {
...
+	e.PATCH("/api/mnemonic", httpSvc.backupReminderHandler, authMiddleware)
```
Comment (rolznz) [inline http_service.go:62]: this is changing the next backup reminder, right? not renaming these fields makes the code very difficult to understand.

File: frontend/src/utils/aesgcm.ts (hunk 21 above)
Comment (rolznz) [reply frontend/src/utils/aesgcm.ts:3]: I can't see the code, or did you write it?

File: frontend/src/screens/apps/AppsList.tsx [hunk 34]
```diff
@@ -1,83 +1,104 @@
...
   return (
-    <>
-      <BreezRedeem />
-      <div className="mb-4 flex justify-between items-center">
-        <h2 className="font-bold text-2xl font-headline dark:text-white">
-          Connected apps
-        </h2>
+    <div className="container max-w-screen-lg mt-6">
+      <div className="flex flex-row-reverse">
+        <BreezRedeem />
```
Comment (rolznz) [inline frontend/src/screens/apps/AppsList.tsx:31]: this should be moved one line up

File: frontend/src/screens/apps/AppsList.tsx (hunk 17 above)
Comment (rolznz) [reply frontend/src/screens/apps/AppsList.tsx:47]: For now, what about just "Create a new connection to connect to an NWC-powered app"

File: frontend/src/screens/apps/ShowApp.tsx [hunk 35]
```diff
@@ -45,105 +101,171 @@ function ShowApp() {
...
+                <td className="text-gray-600 dark:text-neutral-400 break-all">
+                  {app.nostrPubkey}
+                </td>
+              </tr>
+              <tr>
+                <td className="align-top font-medium dark:text-white">
+                  Last used
+                </td>
+                <td className="text-gray-600 dark:text-neutral-400">
+                  {app.lastEventAt &&
```
Comment (rolznz) [inline frontend/src/screens/apps/ShowApp.tsx:162]: Why do we need both checks? Is the date nullable or not?

Ideally this should be nullable, returned from go. Then you don't need this hack to check if the year is 1

File: frontend/src/screens/apps/ShowApp.tsx [hunk 36]
```diff
@@ -45,105 +101,171 @@ function ShowApp() {
...
+              </p>
+            </div>
+            <div className="flex items-center">
+              <div className="sm:w-64 flex-none w-full pt-4 sm:pt-0">
+                <button
+                  type="button"
+                  className="flex-row w-full px-0 py-2 bg-white border border-red-500 text-red-500 dark:bg-surface-02dp dark:text-neutral-200 dark:border-neutral-800 hover:bg-gray-50 dark:hover:bg-surface-16dp cursor-pointer inline-flex justify-center items-center font-medium bg-origin-border shadow rounded-md focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-primary transition duration-150"
+                  onClick={handleDelete}
+                >
+                  Disconnect App
```
Comment (rolznz) [inline frontend/src/screens/apps/ShowApp.tsx:263]: Should this have the same confirmation as on the apps page?

File: frontend/src/screens/setup/ImportMnemonic.tsx [hunk 37]
```diff
@@ -0,0 +1,75 @@
...
+      mnemonic.split(" ").length !== 12 ||
+      !bip39.validateMnemonic(mnemonic, wordlist)
+    ) {
+      toast.error("Invalid recovery phrase");
+      return;
+    }
+
+    setupStore.updateNodeInfo({
+      mnemonic,
+      NextBackupReminder: new Date().toISOString(),
```
Comment (rolznz) [inline frontend/src/screens/setup/ImportMnemonic.tsx:30]: lowercase

File: frontend/src/screens/setup/SetupNode.tsx (hunk 27 above)
Comment (rolznz) [reply frontend/src/screens/setup/SetupNode.tsx:26]: Ah, it's for the import mnemonic flow

File: frontend/src/screens/setup/ImportMnemonic.tsx (hunk 37 above)
Comment (rolznz) [reply frontend/src/screens/setup/ImportMnemonic.tsx:30]: This should be in the future for import. What about 6 months in the future?

File: frontend/src/screens/setup/SetupNode.tsx (hunk 27 above)
Comment (rolznz) [reply frontend/src/screens/setup/SetupNode.tsx:26]: ideally this should be an optional field: `string?` - using the first day of the first year and writing it out manually as an ISO string seems hacky to me

File: frontend/src/screens/setup/SetupNode.tsx [hunk 38]
```diff
@@ -21,13 +23,16 @@ export function SetupNode() {
...
     navigate(
-      backendType === "BREEZ" ||
-        backendType === "GREENLIGHT" ||
-        backendType === "LDK"
-        ? `/setup/mnemonic${isNew ? "?wallet=new" : ""}`
+      !isNew &&
+        (backendType === "BREEZ" ||
+          backendType === "GREENLIGHT" ||
+          backendType === "LDK")
+        ? `/setup/mnemonic`
```
Comment (rolznz) [inline frontend/src/screens/setup/SetupNode.tsx:35]: can you rename the route to `/setup/import-mnemonic`

File: frontend/src/types.ts [hunk 39]
```diff
@@ -78,6 +125,11 @@ export interface InfoResponse {
   setupCompleted: boolean;
   running: boolean;
   unlocked: boolean;
+  NextBackupReminder: boolean;
```
Comment (rolznz) [inline frontend/src/types.ts:128]: casing

File: frontend/src/types.ts [hunk 40]
```diff
@@ -148,6 +200,7 @@ export type SetupNodeInfo = Partial<{
   backendType: BackendType;
 
   mnemonic?: string;
+  NextBackupReminder?: string;
```
Comment (rolznz) [inline frontend/src/types.ts:203]: casing

File: frontend/src/utils/aesgcm.ts (hunk 21 above)
Comment (rolznz) [reply frontend/src/utils/aesgcm.ts:3]: Hmm, I see the code for generating the hash, did you also get the code for the aesGcm encryption/decryption from somewhere?

File: frontend/src/utils/aesgcm.ts [hunk 41]
```diff
@@ -0,0 +1,99 @@
...
+    );
+
+    return [key, salt];
+  } catch (err) {
+    console.error(err);
+    throw new Error("Failed to derive key with Argon2");
+  }
+}
+
+export async function aesGcmEncrypt(
```
Comment (rolznz) [inline frontend/src/utils/aesgcm.ts:40]: This is unused, right? Can we remove it?

File: http_service.go [hunk 42]
```diff
@@ -96,11 +100,33 @@ func (httpSvc *HttpService) csrfHandler(c echo.Context) error {
...
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: err.Error(),
+		})
+	}
 	responseBody.Unlocked = httpSvc.isUnlocked(c)
 	return c.JSON(http.StatusOK, responseBody)
 }
 
+func (httpSvc *HttpService) mnemonicHandler(c echo.Context) error {
```
Comment (rolznz) [inline http_service.go:113]: Please rename this function

File: http_service.go [hunk 43]
```diff
@@ -96,11 +100,33 @@ func (httpSvc *HttpService) csrfHandler(c echo.Context) error {
...
+	responseBody := httpSvc.api.GetEncryptedMnemonic()
+	return c.JSON(http.StatusOK, responseBody)
+}
+
+func (httpSvc *HttpService) backupReminderHandler(c echo.Context) error {
+	currentTime := time.Now()
```
Comment (rolznz) [inline http_service.go:119]: Shouldn't this be passed in through the request?

File: wails_handlers.go [hunk 44]
```diff
@@ -192,10 +209,32 @@ func (app *WailsApp) WailsRequestRouter(route string, method string, body string
...
+			return res
+		case "PATCH":
+			currentTime := time.Now()
+			err := app.api.SetNextBackupReminder(&currentTime)
+			if err != nil {
+				app.svc.Logger.WithFields(logrus.Fields{
+					"route":  route,
+					"method": method,
+					"body":   body,
+				}).Errorf("Failed to backup mnemonic: %v", err)
```
Comment (rolznz) [inline wails_handlers.go:233]: error message is wrong

File: frontend/src/utils/aesgcm.ts (hunk 21 above)
Comment (im-adithya) [reply frontend/src/utils/aesgcm.ts:3]: > I see the code for generating the hash
Yes the rest was just `aesgcm.go` code written in ts

File: frontend/src/screens/apps/AppsList.tsx (hunk 18 above)
Comment (rolznz) [reply frontend/src/screens/apps/AppsList.tsx:74]: I think this can be done, the endpoint needs to be updated to receive the next reminder date through the request anyway

File: api.go (hunk 26 above)
Comment (im-adithya) [reply api.go:670]: It's not empty, it will be passed if the user has entered a mnemonic already (in the case of import), in that case we store the date/time of wallet setup so we can remind them after 6 months (and if it's a new wallet then ofc this would then be set to empty string)

See https://github.com/getAlby/nostr-wallet-connect-next/pull/84/files#diff-45902917f80b550f7a808a08369ae303c3094493a6915d7c785f70b69b7ea5f9R30

File: frontend/src/screens/setup/SetupNode.tsx (hunk 27 above)
Comment (im-adithya) [reply frontend/src/screens/setup/SetupNode.tsx:26]: This is done because 0 time in JS is 1970, but yeah I'll change this to optional field 👍

File: models/api/api.go (hunk 28 above)
Comment (im-adithya) [reply models/api/api.go:60]: It is used right? In the Setup: https://github.com/getAlby/nostr-wallet-connect-next/pull/84/files#diff-f7bb71ca856c07b3329265f26b1e577e00b71e4792550cf14346bdbe0e7bd18dR670

File: frontend/src/components/SuggestedApps.tsx (hunk 30 above)
Comment (im-adithya) [reply frontend/src/components/SuggestedApps.tsx:22]: Changed to SuggestedApp and imported from types ✅

File: frontend/src/utils/aesgcm.ts (hunk 41 above)
Comment (im-adithya) [reply frontend/src/utils/aesgcm.ts:40]: Removed

File: http_service.go (hunk 33 above)
Comment (im-adithya) [reply http_service.go:62]: Changed to `e.PATCH("/api/reminder", httpSvc.backupReminderHandler)`

File: frontend/src/screens/apps/AppsList.tsx (hunk 18 above)
Comment (im-adithya) [reply frontend/src/screens/apps/AppsList.tsx:74]: Set to 6 months ✅

File: api.go (hunk 26 above)
Comment (rolznz) [reply api.go:670]: Cool, thanks!

File: frontend/src/utils/aesgcm.ts (hunk 21 above)
Comment (rolznz) [reply frontend/src/utils/aesgcm.ts:3]: Ah ok, then it should be good. Thanks!

File: models/api/api.go (hunk 28 above)
Comment (rolznz) [reply models/api/api.go:60]: True, thanks!

File: http_service.go (hunk 33 above)
Comment (rolznz) [reply http_service.go:62]: What about `backup-reminder`? there could be other reminders. A reminder on its own again is unclear

File: frontend/src/screens/apps/AppsList.tsx (hunk 18 above)
Comment (rolznz) [reply frontend/src/screens/apps/AppsList.tsx:74]: Wait, this one is to remind me later. I think that's too long. I would do 6 months if the user backs up their wallet, 2 weeks if they skip backing up

File: frontend/src/screens/setup/ImportMnemonic.tsx (hunk 37 above)
Comment (im-adithya) [reply frontend/src/screens/setup/ImportMnemonic.tsx]: Done!

File: http_service.go (hunk 43 above)
Comment (im-adithya) [reply http_service.go]: Done!

File: frontend/src/components/AppCard.tsx (hunk 29 above)
Comment (im-adithya) [reply frontend/src/components/AppCard.tsx]: Done!

File: frontend/src/screens/apps/ShowApp.tsx (hunk 36 above)
Comment (im-adithya) [reply frontend/src/screens/apps/ShowApp.tsx]: Added
